protocol/src/main/protobuf/UserBitShared.proto

```diff
@@ -298,3 +298,17 @@ enum CoreOperatorType {
   NESTED_LOOP_JOIN = 35;
   AVRO_SUB_SCAN = 36;
 }
+
+message Func {
+  optional string name = 1;
+  repeated common.MajorType major_type = 2;
+}
+
+message Jar {
+  optional string name = 1;
+  repeated Func function = 2;
+}
+
+message Registry {
```

Comment:
When do we exchange a registry of Jars? Is this stored in ZK? If so, do we have to worry about the total size of the resulting ZK zNode? Have we done performance checks to see what happens if the user has registered, say, 100 jars of 5 functions each? Should we consider a ZK structure (if that is what this is for)  of the form:

.../udfs
  jar1: <JAR>
  jar2: <JAR>

That is, store the list of jars as zNodes, where each zNode defines one jar?